### PR TITLE
Make functions fit for use of GCMS data

### LINKED ISSUE #83: Preparations for GC MS
We will soon try using matchms on GC MS data.
From what we can see already it might be good to define 2 different ``default_filters``.
E.g.:
- ``default_filters_lcms``
- ``default_filters_gcms`` (not using any filters that rely on precursor_mz or parent_mass)

### PR DESCRIPTION
**Pull request by florian-huber**
- raise exception when ``reduce_to_number_of_peaks`` is used with ``desired_ratio`` but without parent mass (e.g. for GC MS data). Before it would then just ignore the parameter, but that can be very unexpected behavior.
- fix small issue with msp importer function

### CODE REVIEW

File: matchms/filtering/reduce_to_number_of_peaks.py [hunk 1]
```diff
@@ -19,14 +19,18 @@ def reduce_to_number_of_peaks(spectrum_in: SpectrumType, n_required: int = 1, n_
...
     """
     def _set_maximum_number_of_peaks_to_keep():
         parent_mass = spectrum.get("parent_mass", None)
         if parent_mass and ratio_desired:
             n_desired_by_mass = int(ceil(ratio_desired * parent_mass))
             return max(n_required, n_desired_by_mass)
-        return n_max
+        if not ratio_desired:
+            return n_max
+        raise ValueError("Cannot use ratio_desired for spectrum without parent_mass.")
```
Comment (sverhoeven) [inline matchms/filtering/reduce_to_number_of_peaks.py:33]: Can you add a test for when ratio_desired=None and spectrum has no parent.

File: matchms/importing/load_from_msp.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ def parse_msp_file(filename: str) -> Generator[dict, None, None]:
     # Peaks counter. Used to track and count the number of peaks
     peakscount = 0
 
-    with open(filename, 'r') as f:
+    with open(filename, 'r', errors='replace') as f:
```
Comment (sverhoeven) [inline matchms/importing/load_from_msp.py]: To me it is dangerous to replace things while reading the users input file without notifying the user. 
Do you have an example where this was needed?

**Review by sverhoeven — CHANGES_REQUESTED** [review]

File: matchms/importing/load_from_msp.py (hunk 2 above)
Comment (florian-huber) [reply matchms/importing/load_from_msp.py]: Yes. I got an unicode error when I tried to re-run a notebook from Efrain (student of Simon).
Here is the notebook: https://github.com/florian-huber/spec2vec_gcms_data_analysis/blob/master/notebooks/FH_re_run_pipeline_01.ipynb
The file (also present in the repo) is "MoNA-export-GC-MS.msp".

Comment (sverhoeven) [reply matchms/importing/load_from_msp.py]: Hmm, on a cloned repo I was able to read the file with `f = open('MoNA-export-GC-MS.msp').read()` with ipython 7.18.1 Python 3.8.5 n Ubuntu 20.04 using

```python

In [1]: 
   ...: import os
   ...: import sys
   ...: import numpy as np

In [2]: msp_file ="MoNA-export-GC-MS.msp"

In [3]: from matchms.filtering import normalize_intensities
   ...: from matchms.filtering import reduce_to_number_of_peaks
   ...: from matchms.filtering import require_minimum_number_of_peaks
   ...: from matchms.filtering import select_by_mz
   ...: from matchms.importing import load_from_msp

In [4]: z = list(load_from_msp(msp_file))

In [5]: len(z)
Out[5]: 14847

```
I did not get an unicode error, so I am unable to reproduce.

Comment (florian-huber) [reply matchms/importing/load_from_msp.py]: I had it run on a Windows machine and get
``UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 135: character maps to <undefined>``

Comment (florian-huber) [reply matchms/importing/load_from_msp.py]: I changed it to ``open(filename, 'r', encoding='utf-8')`` (thought I'd tested that before, but now seems to run fine).

File: matchms/filtering/reduce_to_number_of_peaks.py (hunk 1 above)
Comment (florian-huber) [reply matchms/filtering/reduce_to_number_of_peaks.py:33]: Added a test case for ratio_desired=None but with parent mass (the one without both is already present since default is None).

**Review by sverhoeven — APPROVED** [review]
OK, encoding should work for ascii and utf

**Comment by florian-huber** [comment]
Thanks a lot Stefan for reviewing!
